src/modules/announcements/pages/Announcements.jsx

```diff
@@ -1,3 +1,53 @@
+import { useEffect, useState } from "react";
+import { useLocation } from "react-router-dom";
+import { SearchForm } from "../../../components/SearchForm/SearchForm";
+import { ANNOUNCEMENTS_LIST } from "../../../constants/announcements";
+
 export const AnnouncementList = () => {
-  return <div>announcement list</div>;
+  const [filteredAnnouncements, setFilteredAnnouncements] = useState([]);
+  const { search } = useLocation();
+
+  const filterAnnouncements = (data) => {
+    const { Phrase, Category, City, Animals } = data;
+    let announcements = [...ANNOUNCEMENTS_LIST];
+    let animalsArray;
+
+    announcements = announcements
+      .filter((announcement) =>
+        announcement.title.toLowerCase().includes(Phrase.toLowerCase()),
+      )
+      .filter((announcement) =>
+        announcement.category.toLowerCase().includes(Category),
+      )
+      .filter((announcement) =>
+        announcement.city.toLowerCase().includes(City.toLowerCase()),
+      );
+    if (Animals) {
+      animalsArray = Animals.map((animal) => animal.toLowerCase());
+      announcements = announcements.filter((announcement) =>
+        animalsArray.includes(announcement.animal.toLowerCase()),
+      );
+    }
+
+    setFilteredAnnouncements(announcements);
+  };
+  useEffect(() => {
+    if (search) {
+      const params = new URLSearchParams(search);
+      const Phrase = params.get("phrase");
+      const Category = params.get("category");
+      const City = params.get("city");
+      const Animals = params.getAll("animal");
+      filterAnnouncements({ Phrase, Category, City, Animals });
+    } else {
+      setFilteredAnnouncements(ANNOUNCEMENTS_LIST);
+    }
+  }, []);
+  console.log(filteredAnnouncements);
```

Comment:
Remove it.

Reply:
I can remove it but eslint will throw error that variable is not used. Everyone who create new branch based on this will have to fix this error

Reply:
Sorry, I don' get it still. If it is not used then it is not used. Is there any reason why `eslint` shouldn't detect it as unused if it is unused? Just put ATM in the state `const [_, setfilteredAnnouncements]` if that is the reason

Reply:
still eslint will throw error. "_" is unused etc...